pkg/accesscontrolrule/helper.go

```diff
@@ -337,6 +341,12 @@ func (h *Helper) UserHasAccessToMCPServerInWorkspace(user kuser.Info, serverName
 		userID = user.GetUID()
 		groups = authGroupSet(user)
 	)
+
+	// If the server is owned by the current user, they have access to it to ignore the AccessControlRules
+	if serverUserID == userID {
+		return true, nil
+	}
```

Comment:
Why not do this as the very first thing in the function?

Reply:
no reason than being dumb, move :)